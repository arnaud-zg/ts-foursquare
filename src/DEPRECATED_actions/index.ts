import { TLifeAction } from './life'

export type TRootAction = TLifeAction

export * from './life'
