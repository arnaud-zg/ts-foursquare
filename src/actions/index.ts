import { TLifeAction } from './life'
import { TVenuesAction } from './venues'

export type TRootActionType = TVenuesAction | TLifeAction

export type TRootAction = TVenuesAction | TLifeAction

export * from './life'
export * from './venues'
