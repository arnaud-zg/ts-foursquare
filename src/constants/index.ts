export const PROJECT_NAME = 'ts-foursquare'
