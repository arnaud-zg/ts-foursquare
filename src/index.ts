require('isomorphic-fetch')

export * from './standalone'
export * from './utils'

export * from './DEPRECATED_actions'
export * from './DEPRECATED_reducers'
export * from './DEPRECATED_selectors'
