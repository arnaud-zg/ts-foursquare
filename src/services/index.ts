export * from './lists'
export * from './photos'
export * from './venues'
