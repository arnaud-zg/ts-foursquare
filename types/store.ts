import { NLife } from './lifeState'
import { NStatus } from './statusState'
import { NVenues } from './venuesState'

export declare namespace NStore {
  interface IState {
    life: NLife.IState
    status: NStatus.IState
    venues: NVenues.IState
  }
}
