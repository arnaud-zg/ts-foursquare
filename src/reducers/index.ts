import { NVenues } from '../../types/venues'

export type TRootState = {
  venues: NVenues.IVenuesState
}
