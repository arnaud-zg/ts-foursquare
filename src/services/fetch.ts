import { StateObservable } from 'redux-observable'
import { Observable } from 'rxjs/internal/Observable'
import { throwError } from 'rxjs/internal/observable/throwError'
import { NStore } from '../../types/store.d'
import { EApiDefaultParameters } from '../constants/api'
import { credentialsSelector } from '../selectors/life'

export const processFetchResponse = (
  response: Response
): Promise<any> | Observable<Error> => {
  if (response.ok) {
    return response.json()
  }

  return throwError(new Error(`Error ${response.status}`))
}

export const processFetchError = (err: Error): Observable<Error> =>
  throwError(new Error(err.message))

export const getDefaultRequestParameters = (
  state$: StateObservable<NStore.IState>
) => {
  const { clientId, clientSecret } = credentialsSelector(state$.value)

  return {
    client_id: clientId,
    client_secret: clientSecret,
    v: EApiDefaultParameters.VERSION,
  }
}
