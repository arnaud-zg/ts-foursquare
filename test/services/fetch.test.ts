import { getDefaultRequestParameters } from '../../src/services/fetch'
import { StateObservable } from 'redux-observable'
import { NStore } from '../../types/store'
import { initialState } from '../../src/reducers'
import { Subject } from 'rxjs'

describe('utils/fetch/getDefaultRequestParameters', () => {
  it('should get default request parameters', () => {
    const state$: StateObservable<NStore.IState> = new StateObservable(
      new Subject(),
      {
        ...initialState,
        life: {
          ...initialState.life,
          credentials: {
            ...initialState.life.credentials,
            clientId: '123',
            clientSecret: '456',
          },
        },
      }
    )

    expect(getDefaultRequestParameters(state$)).toMatchSnapshot()
  })
})
