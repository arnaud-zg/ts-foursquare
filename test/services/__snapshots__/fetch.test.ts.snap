// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils/fetch/getDefaultRequestParameters should get default request parameters 1`] = `
Object {
  "client_id": "123",
  "client_secret": "456",
  "v": "20191007",
}
`;
